gossip_test.go

```diff
@@ -66,6 +67,8 @@ func sendPendingGossip(routers ...*Router) {
 			sentSomething = router.sendPendingGossip() || sentSomething
 		}
 	}
+	// Allow topology to propagate
+	time.Sleep(1 * time.Second)
```

Comment:
I don't like this - `sendPendingGossip()` is carefully written to wait long enough already.

You could add a new function in `peers_test.go` like:
```go
func forcePendingGC(routers ...*Router) {
	for _, router := range routers {
		router.Peers.Lock()
		if router.Peers.pendingGC {
			var pending peersPendingNotifications
			router.Peers.garbageCollect(&pending)
			router.Peers.unlockAndNotify(&pending)
		} else {
			router.Peers.Unlock()
		}
	}
}
```

then call it as necessary?